### PR TITLE
Support beam_refactor on pangeo_forge_recipes

### PR DESCRIPTION
**Pull request by yuvipanda**
pangeo_forge_recipes still needs to find a way to get the storage options into it in beam_refactor though.

### CODE REVIEW

File: pangeo_forge_runner/commands/bake.py [hunk 1]
```diff
@@ -155,7 +148,22 @@ def start(self):
                 # for pipeline options - we have traitlets doing that for us.
                 pipeline = Pipeline(options=pipeline_options, argv=[])
                 # Chain our recipe to the pipeline. This mutates the `pipeline` object!
-                pipeline | recipe.to_beam()
+                # We expect `recipe` to either be a beam PTransform, or an object with a 'to_beam'
+                # method that returns a transform.
+                if isinstance(recipe, PTransform):
+                    # This means we are in pangeo-forge-recipes >=0.9
```
Comment (cisaacstern) [inline pangeo_forge_runner/commands/bake.py:154]:
```suggestion
                    # This means we are in pangeo-forge-recipes >=0.10
```

**Comment by cisaacstern** [comment]
This looks awesome, @yuvipanda! And it's a testament to the initial application design that so little needed to be changed.

I propose we leave this open until we can get a `beam-refactor` version of https://github.com/pforgetest/gpcp-from-gcs-feedstock created (ideally today or tomorrow) to test it against. Does that make sense?

**Comment by cisaacstern** [comment]
> Edit: I will start working on that test feedstock now.

See https://github.com/pforgetest/gpcp-from-gcs-feedstock/pull/4

**Comment by cisaacstern** [comment]
@yuvipanda I opened #49 against this PR to incorporate testing against https://github.com/pforgetest/gpcp-from-gcs-feedstock/pull/4.

`Unit Tests` workflow succeeded on the last commit here, but I expect it will fail for the current commit of #49, because of the `FIXME` items in https://github.com/pforgetest/gpcp-from-gcs-feedstock/pull/4. Hopefully this test failure will point us towards next steps.

**Comment by yuvipanda** [comment]
The problem with flink here is unrelated to beam refactor, as I managed to get beam running on flink on EKS - so hmmmmm

**Comment by yuvipanda** [comment]
Flink setup is unrelated, going to merge now :)

**Pull request merged**